pkg/minikube/config/profile.go

```diff
@@ -206,8 +207,9 @@ func ListProfiles(miniHome ...string) (validPs []*Profile, inValidPs []*Profile,
 	if err == nil {
 		pDirs = append(pDirs, cs...)
```

Comment:
the source of the problem is this ListOwnedContainers, that returns "all continuers" including multi node ones
for example it returns

```
[ "minikube",  "minikube-m02"]
```

we could add a check here, to append only if it is not a node conaterin

```
	// try to get profiles list based on all containers created by docker driver
	cs, err := oci.ListOwnedContainers(oci.Docker)
	if err == nil {
                   // add logic here to check if it is primary container or worker node container
		pDirs = append(pDirs, cs...)
	}
```

Reply:
additionaly we can make the oci.ListOwnedContainers(oci.Docker) to have an option to only return the Master Containers

Reply:
I agree fix `ListOwnedContainers` is the best way.
Could you let me know how can i filter nodes?

Reply:
unfornatunately in multi node implementaiotn we didnt use the labels https://githu
b.com/medyagh/minikube/blob/9e5beb4e37298d296694cd7339740e336c62a653/pkg/drivers/kic/oci/types.go#L39

so we can't filter the nodes by master there ... ( we could do that for future releases though)
but one simple thing we can do is filter out any -m01 -m02 (digits) from the returned list
so it only returns minikube and not minikbe-m02

Reply:
hmm I'm worrying about user make profile with -m02 postfix..
```
minikube start -p test-profile-m02
```

Reply:
Filter out any -m0x node will have a potential false-negative result with profile.
Do we have any other way??